inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -700,21 +704,25 @@ std::shared_ptr<CNNNetworkImpl> convertFunctionToICNNNetwork(const std::shared_p
     // Collect all names from current graph
     // It is necessary in order to differentiate outputs from constant layers when we share constants
     // (Constant operations contains outputs for converted and original functions)
-    const ngraph::NodeVector& nodes = graph->get_ops();
+    const ngraph::NodeVector &nodes = graph->get_ops();
 
     std::unordered_set<std::string> op_names;
     for (const auto &layer : nodes)
         op_names.insert(layer->get_name());
 
-    bool keep_constants = ::ngraph::op::util::has_op_with_type<::ngraph::op::FakeQuantize>(graph);
-
+    bool keep_constants;
+    if (keep_constant_inputs == false) {
+        keep_constants = ::ngraph::op::util::has_op_with_type<::ngraph::op::FakeQuantize>(graph);
+    } else {
+        keep_constants = keep_constant_inputs;
+    }
```

Comment:
Why not just

```
bool keep_constants = keep_constant_inputs || ::ngraph::op::util::has_op_with_type<::ngraph::op::FakeQuantize>(graph);
```

As Ilya said, `FakeQuantize` operation presence should force the constant keep. The `keep_constant_inputs` parameter is just an option for plugin to disable the constants fusing for non-quantized networks.

Reply:
Your concise code snippet is used for substitution.